### PR TITLE
new command hw create-credential

### PR DESCRIPTION
**Pull request by caberos**
[#issue1677](https://github.com/softlayer/softlayer-python/issues/1773)

image

### CODE REVIEW

File: SoftLayer/CLI/hardware/create_credential.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    software = mgr.get_software_components(identifier)
+    sw_id = None
+    for sw_instance in software:
+        if (sw_instance['softwareLicense']['softwareDescription']['name']).lower() == system:
+            sw_id = sw_instance['id']
+            break
+        elif system:
+            if (sw_instance['softwareLicense']['softwareDescription']['name']) == 'Passmark Suite':
+                sw_id = sw_instance['id']
```
Comment (allmightyspiff) [inline SoftLayer/CLI/hardware/create_credential.py:31]: I dont understand what you are checking for here?

Looks to me like you are checking for if a user picked `--system`, but then you just set sw_id to `Passmark Suite`. I feel like this whole `elif` check could be removed, and if at the end if your for loop, if `sw_id` is None and `system` is not None, throw and exception about not being able to find a software called `system`

File: SoftLayer/managers/hardware.py [hunk 2]
```diff
@@ -1102,6 +1102,14 @@ def add_notification(self, hardware_id, user_id):
         template = {"hardwareId": hardware_id, "userId": user_id}
         return self.client.call('SoftLayer_User_Customer_Notification_Hardware', 'createObject', template)
 
+    def get_software_components(self, hardware_id):
+        """Returns Hardware sensor data"""
```
Comment (allmightyspiff) [inline SoftLayer/managers/hardware.py:1106]: Fix this doc line to reflect what this function actually does.

File: SoftLayer/CLI/hardware/create_credential.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+
+
+@click.command(cls=SoftLayer.CLI.command.SLCommand, )
+@click.argument('identifier')
+@click.option('--username', '-U', help="The username part of the username/password pair")
+@click.option('--password', '-P', help="The password part of the username/password pair.")
+@click.option('--notes', '-n', help="A note string stored for this username/password pair.")
+@click.option('--system', help="The name of this specific piece of software.")
```
Comment (allmightyspiff) [inline SoftLayer/CLI/hardware/create_credential.py:16]: If these are required, you should mark them as such, and if not, set a default value.

**Review by allmightyspiff — CHANGES_REQUESTED** [review]

**Review by edsonarios — APPROVED** [review]
Looks good to me

**Review by allmightyspiff — APPROVED** [review]

**Pull request merged**